### DISCUSSION TITLE
Understanding useAuth and useUser in saleor dashboard

### DISCUSSION THREAD

**Discussion by crisfcodes** [saleor/saleor#8740] [Q&A] [answered]
What's the difference between `useAuth` and `useUser` in saleor-dashboard... both use `UserContext`... `useAuth` has some extra properties but why have two context almost identical? the other thing is, I think from what I see in the folder layout, that you are running a feature based architecture... are these features not supposed to be isolated from each other or what criteria do you use to know where to share things?

**Comment by jwm0** [comment, marked as answer]
Hey @crisfcodes, to answer your first question... there's no difference - I've just refactored the code to use `useUser` everywhere (which seems more appropriate for context named UserContext), nice catch! Our folder structure is based on a feature/module architecture but it's more of a decision for us to keep things organized and isolated from each other. Components with business logic and very specific UIs tend to land in the feature/module directory as that's the only place they'll be used, whereas more generic ones that can be reused in multiple modules are usually found in the `components/` folder. I'm sure you'll find places where this may not be true as the project is evolving and more people are working on it but that's the general architecture and it should be true for most of the codebase and how we want to proceed in the future.
